### ISSUE TITLE
[clang-format] Add support for C language to IntegerLiteralSeparator

### ISSUE THREAD

**Issue by JWTheDBA** [llvm/llvm-project#179686]
C23 adopted the C++ style tick mark for numeric separators.  Allow clang-format IntegerLiteralSeparator to function when Language: C is selected.

**Comment by JWTheDBA** [comment]
@owenca I think this issue was misunderstood and should be re-opened.  I was not reporting a formatting issue with the IntegerLiteralSeparator but was asking that the setting function when Language: C is selected.  Per the documentation, and my own testing, this setting applies only for C++, C#, Java, and Javascript.


.clang-format
Language: C 
IntegerLiteralSeparator:
  Binary: 4
    
Before formatting:
```c
char data = 0b10000110;

int
main(void) {
        return 0;
}
```
After formatting:
```c
char data = 0b10000110;

int
main(void) {
        return 0;
}
```

.clang-format
Language: Cpp
IntegerLiteralSeparator:
  Binary: 4    

```c
char data = 0b1000'0110;

int
main(void) {
        return 0;
}
```

As originally stated now that C23 supports the ' separator please enable the feature for Language: C

**Comment by fauxprogrammer** [comment]
@owenca Are you able to review and approve if correct?
